From 3bc74b339fb8bc0574290769c799b37dde6b890a Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Tue, 31 May 2011 16:52:15 +0800
Subject: [PATCH 986/998] ENGR00144287-1 MSL: make hdmi work for both mx53 and mx50

After mx50 hdmi patches in, it make issues of mx53 platform.
This patch fix such issues on mx53 like:
1. dual display will make edid video mode into both fb0 and fb1
2. after v4l2 output playback, fb0 will be blank.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 arch/arm/mach-mx5/mx50_rdp.c  |    1 +
 arch/arm/mach-mx5/mx53_evk.c  |    1 +
 arch/arm/mach-mx5/mx53_loco.c |    1 +
 arch/arm/mach-mx5/mx53_smd.c  |    1 +
 4 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx5/mx50_rdp.c b/arch/arm/mach-mx5/mx50_rdp.c
index 4371336..46b33d8 100644
--- a/arch/arm/mach-mx5/mx50_rdp.c
+++ b/arch/arm/mach-mx5/mx50_rdp.c
@@ -1284,6 +1284,7 @@ static struct mxc_lcd_platform_data sii902x_hdmi_data = {
 	.put_pins = sii902x_hdmi_put_pins,
 	.enable_pins = sii902x_hdmi_enable_pins,
 	.disable_pins = sii902x_hdmi_disable_pins,
+	.fb_id = "mxc_elcdif_fb",
 };
 
 static struct i2c_board_info mxc_i2c1_board_info[] __initdata = {
diff --git a/arch/arm/mach-mx5/mx53_evk.c b/arch/arm/mach-mx5/mx53_evk.c
index 3afa5f9..b0571fa 100644
--- a/arch/arm/mach-mx5/mx53_evk.c
+++ b/arch/arm/mach-mx5/mx53_evk.c
@@ -865,6 +865,7 @@ static void sii902x_hdmi_reset(void)
 
 static struct mxc_lcd_platform_data sii902x_hdmi_data = {
 	.reset = sii902x_hdmi_reset,
+	.fb_id = "DISP3 BG",
 };
 
 static void ddc_dvi_init(void)
diff --git a/arch/arm/mach-mx5/mx53_loco.c b/arch/arm/mach-mx5/mx53_loco.c
index f0569d4..8b5d4cf 100644
--- a/arch/arm/mach-mx5/mx53_loco.c
+++ b/arch/arm/mach-mx5/mx53_loco.c
@@ -438,6 +438,7 @@ static void sii902x_hdmi_reset(void)
 
 static struct mxc_lcd_platform_data sii902x_hdmi_data = {
        .reset = sii902x_hdmi_reset,
+       .fb_id = "DISP3 BG",
 };
 
 static struct imxi2c_platform_data mxci2c_data = {
diff --git a/arch/arm/mach-mx5/mx53_smd.c b/arch/arm/mach-mx5/mx53_smd.c
index d8e0695..042ec5f 100644
--- a/arch/arm/mach-mx5/mx53_smd.c
+++ b/arch/arm/mach-mx5/mx53_smd.c
@@ -615,6 +615,7 @@ static void sii902x_hdmi_reset(void)
 static struct mxc_lcd_platform_data sii902x_hdmi_data = {
 	.reset = sii902x_hdmi_reset,
 	.analog_reg = "DA9052_LDO2",
+	.fb_id = "DISP3 BG",
 	.boot_enable = 1,
 };
 
-- 
1.7.4.1

